### DISCUSSION TITLE
Doubt regarding dev mode

### DISCUSSION THREAD

**Discussion by antropolouco** [pretix/pretix#4539] [Q&A]
Hello there!

Tried email support, but I'm not paying enterprise etc, so they sent me here.

I did a Pretix development setup, and so far, so good. Now I'm wondering if there's any chance I can put it to production. For what I'm seeing, it seems I would have to make a whole different install if I want Pretix out of the development mode, so I'm writing you to confirm.

Is that the case? If it is, I ask you: what are my restrictions while in development mode? Can I still use Pretix in dev mode for managing a single event without worries (with manual payment processing & qr code check-in), for example?

Thank you very much for your work and support.

**Comment by raphaelm** [comment]
The development setup is only for developing pretix locally on your machine. You absolutely should not use it for your production event. It may have performance or security issues.

**Reply by antropolouco** [reply]
Well, thanks but that did not answer at all, I guess. It's just a vague recommendation - but it should not be used for production, I agree.

I'm trying it, and dev is ok for my needs now. I can witness that dev mode can work fine for small events and for what I need now, so if it's an easier install people can know that's possible for some use cases (even assuming such performance and security risks).

Best regards

**Reply by raphaelm** [reply]
I believe it does answer your question:

> Is that the case? If it is, I ask you: what are my restrictions while in development mode? 

There are performance restrictions (due to not using a proper web server) and possible security gaps.

> Can I still use Pretix in dev mode for managing a single event without worries (with manual payment processing & qr code check-in), for example?

Yes, you can used it but I would not say "without worries", as these are issues I'd be worried about.

Additionally, it leaves you with a setup that only you have and people will not know how to help you in situations like #4540 because your setup is unique. You can still figure it all out, for sure.

**Reply by antropolouco** [reply]
Well, it does not answer still, in my opinion. The only detail you provide is linking to the CSRF issue I opened, and repeating vague statements ("performance" and "security" are thrown in the air and that's it), so it won't be helpful for anyone with this kind of doubt. At least somebody can reach the info that CSRF is "probably not impossible". The only conclusion I can reach is that the dev install is much more functional than you are willing to recognize.

But thanks for you work!

Best regards.

**Reply by felixrindt** [reply]
With your apparent knowledge and attitude towards this topic you should not be setting up software that's handling user data. 
Please read up on security in hosting (Django) applications. [Start here](https://docs.djangoproject.com/en/5.1/ref/settings/#debug).

**Reply by antropolouco** [reply]
Yay... Sure, my attitude. You know me pretty well, hum? Yeah, whatever.

Questioning poor (in my case, notably unwilling) support is not bad. I hope you (all) don't take it personally, and have a very good day.

Thanks, Felix!
